Alignment/OfflineValidation/macros/FitPVResiduals.C

```diff
@@ -0,0 +1,2151 @@
+#include "TFile.h"
+#include "TPaveStats.h"
+#include "TROOT.h"
+#include "TList.h"
+#include "TNtuple.h"
+#include "TTree.h"
+#include "TH1.h"
+#include "TArrow.h"
+#include "TH2.h"
+#include "THStack.h"
+#include "TStyle.h"
+#include "TLegendEntry.h"
+#include "TPaveText.h"
+#include "TCut.h"
+#include "TLegend.h"
+#include "TGraphErrors.h"
+#include "TF1.h"
+#include "TMath.h"
+#include "TVectorD.h"
+#include "TCanvas.h"
+#include "TColor.h"
+#include "TAxis.h"
+#include "TGaxis.h"
+#include "TROOT.h"
+#include "TObjArray.h"
+#include "TGraphErrors.h"
+#include "TF1.h"
+#include "TMinuit.h"
+#include "TString.h"
+#include "TMath.h"
+#include <TDatime.h>
+#include <TSpectrum.h>
+#include <TSystem.h>
+#include <TStopwatch.h>
+#include "Alignment/OfflineValidation/plugins/TkAlStyle.cc" 
+#include "CMS_lumi.C"
+
+#define ARRAY_SIZE(array) (sizeof((array))/sizeof((array[0])))
+
+void arrangeCanvas(TCanvas *canv,TH1F* meanplots[100],TH1F* widthplots[100],Int_t nFiles,TString LegLabels[10],TString theDate="bogus",bool onlyBias=false);
+void arrangeCanvas2D(TCanvas *canv,TH2F* meanmaps[100],TH2F* widthmaps[100],Int_t nFiles,TString LegLabels[10],TString theDate="bogus");
+void arrangeFitCanvas(TCanvas *canv,TH1F* meanplots[100],Int_t nFiles, TString LegLabels[10],TString theDate="bogus");
+
+void arrangeBiasCanvas(TCanvas *canv,TH1F* dxyPhiMeanTrend[100],TH1F* dzPhiMeanTrend[100],TH1F* dxyEtaMeanTrend[100],TH1F* dzEtaMeanTrend[100],Int_t nFiles, TString LegLabels[10],TString theDate="bogus");
+
+std::pair<Double_t,Double_t> getMedian(TH1F *histo);
+std::pair<Double_t,Double_t> getMAD(TH1F *histo);
+
+std::pair<std::pair<Double_t,Double_t>, std::pair<Double_t,Double_t>  > fitResiduals(TH1 *hist);
+
+Double_t DoubleSidedCB(double* x, double* par);
+std::pair<std::pair<Double_t,Double_t>, std::pair<Double_t,Double_t>  > fitResidualsCB(TH1 *hist);
+
+Double_t tp0Fit( Double_t *x, Double_t *par5 );
+std::pair<std::pair<Double_t,Double_t>, std::pair<Double_t,Double_t>  > fitStudentTResiduals(TH1 *hist);
+
+void FillTrendPlot(TH1F* trendPlot, TH1F* residualsPlot[100], TString fitPar_, TString var_,Int_t nbins);
+void FillMap(TH2F* trendMap, TH1F* residualsMapPlot[48][48], TString fitPar_);
+
+void MakeNiceTrendPlotStyle(TH1 *hist,Int_t color);
+void MakeNicePlotStyle(TH1 *hist);
+void MakeNiceMapStyle(TH2 *hist);
+void MakeNiceTF1Style(TF1 *f1,Int_t color);
+
+void FitPVResiduals(TString namesandlabels,bool stdres=true,bool do2DMaps=false,TString theDate="bogus");
+TH1F* DrawZero(TH1F *hist,Int_t nbins,Double_t lowedge,Double_t highedge,Int_t iter);
+void makeNewXAxis (TH1F *h);
+void makeNewPairOfAxes (TH2F *h);
+
+// ancillary fitting functions
+Double_t fULine(Double_t *x, Double_t *par);
+Double_t fDLine(Double_t *x, Double_t *par);
+void FitULine(TH1 *hist);
+void FitDLine(TH1 *hist);
+
+void setStyle();
+
+// global variables
+
+ofstream outfile("FittedDeltaZ.txt");
+Int_t my_colors[10]={kBlack,kRed,kBlue,kMagenta,kBlack,kRed,kBlue,kGreen};
```

Comment:
@mmusich the number of colours on the rhs is 8, i.e. the remaining colours are filled up with 0
Is this intended?